src/components/Sing/ScoreSequencer.vue

```diff
@@ -1,23 +1,348 @@
+            @dblclick="removeNote(index)"
+            class="sequencer-note-bar"
+          />
+        </svg>
+      </div>
+    </div>
+    <!-- NOTE: スクロールバー+ズームレンジ仮 -->
+    <input
+      type="range"
+      min="0.2"
+      max="1"
+      step="0.1"
+      :value="zoomX"
+      @change="(e) => setZoomX(e)"
+      v-bind:style="{
+        position: 'fixed',
+        zIndex: 10000,
+        bottom: '8px',
+        right: '16px',
+        width: '80px',
+      }"
+    />
+    <input
+      type="range"
+      min="0.2"
+      max="1"
+      step="0.1"
+      :value="zoomY"
+      @change="(e) => setZoomY(e)"
+      v-bind:style="{
+        position: 'fixed',
+        zIndex: 10000,
+        bottom: '64px',
+        right: '-8px',
+        transform: 'rotate(-90deg)',
+        width: '80px',
+      }"
+    />
   </div>
 </template>
 
 <script lang="ts">
-import { defineComponent } from "vue";
+import { defineComponent, computed, onMounted } from "vue";
+import { useStore } from "@/store";
+import {
+  midiKeys,
+  getPitchFromMidi,
+  getDoremiFromMidi,
+} from "@/helpers/singHelper";
 
 export default defineComponent({
   name: "SingScoreSequencer",
-
   setup() {
-    return null;
+    const store = useStore();
+    const gridY = midiKeys;
+    const gridX = computed(() => {
+      const resolution = store.state.score?.resolution || 480;
+      // NOTE: 最低幅: 仮16小節
+      const minDuration = resolution * 4 * 16;
+      const lastNote = store.state.score?.notes.slice(-1)[0];
+      const totalDuration = lastNote
+        ? Math.max(lastNote.position + lastNote.duration, minDuration)
+        : minDuration;
+      // NOTE: グリッド幅1/16: 設定できるようにする必要あり
+      const gridDuration = resolution / 4;
+      return [...Array(Math.ceil(totalDuration / gridDuration)).keys()].map(
+        (gridNum) => gridNum * gridDuration
+      );
+    });
+    const notes = computed(() => store.state.score?.notes);
+    const timeSignatures = computed(() => store.state.score?.timeSignatures);
+    const zoomX = computed(() => store.state.sequencerZoomX);
+    const zoomY = computed(() => store.state.sequencerZoomY);
+    const scrollX = computed(() => store.state.sequencerScrollX);
+    const scrollY = computed(() => store.state.sequencerScrollY);
+    const BASE_Y_SIZE = 30;
+    const BASE_X_SIZE = 30;
+    onMounted(() => {
+      const el = document.querySelector("#score-sequencer");
+      // C4あたりにスクロールする
+      if (el) {
+        el.scrollTop = scrollY.value * (BASE_Y_SIZE * zoomY.value);
+      }
+    });
+    const addNote = (event: PointerEvent) => {
+      const snapSize = store.state.sequencerSnapSize;
+      const resolution = store.state.score?.resolution;
+      const gridXSize = resolution ? resolution / 4 : 120;
+      const position =
+        gridXSize * Math.floor(event.offsetX / (BASE_X_SIZE * zoomX.value));
+      const midi =
+        127 - Math.floor(event.offsetY / (BASE_Y_SIZE * zoomY.value));
+      const duration = snapSize;
```

Comment:
Y軸の拡大率を最小にして、ピアノロールのC-1より下の方をクリックしたときなど、zoomYの値が小さいときにmidiの値がマイナスになってしまう場合があるので、マイナスになっている場合はノートを追加しないようにした方が良いかなと思います！
また、action内でもノートの値をチェックするようにした方が良いかもしれません。
```suggestion
      const midi =
        127 - Math.floor(event.offsetY / (BASE_Y_SIZE * zoomY.value));
      if (midi < 0) return;
      const duration = snapSize;
```

Reply:
@sigprogramming 
ありがとうございます、おっしゃる状況たしかにありえそうです！
(いただいたものそのままコミットさせていただきました！)